src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -32,6 +31,18 @@
 {
   private final KeyedDiff gcDiff = new KeyedDiff();
 
+  private Map<String, String[]> dimensions;
+
+  public JvmMonitor()
+  {
+    this(ImmutableMap.<String, String[]>of());
+  }
+
+  public JvmMonitor(Map<String, String[]> dimensions)
+  {
+    this.dimensions = ImmutableMap.copyOf(dimensions);
```

Comment:
we should to a `Preconditions.checkNotNull` on dimensions here to guard against nulls